src/appleseed.python/bindblendermeshconverter.cpp

```diff
@@ -0,0 +1,222 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Jonathan Dent, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.renderer headers.
+#include "renderer/api/object.h"
+
+// appleseed.foundation headers.
+#include "foundation/platform/python.h"
+
+namespace bpy = boost::python;
+using namespace renderer;
+
+// Blender data structures
+struct MeshFace
+{
+    unsigned int v[4];
+    short mat_nr;
+    char edcode, flag;
+};
+
+struct MeshVert
+{
+    float co[3];
+    short no[3];
+    char flag, bweight;
+};
+
+struct MeshTexFace
+{
+    float uv[4][2];
+    void *tpage;
+    char flag, transp;
+    short mode, tile, unwrap;
+};
+
+
+// The following function takes a series of pointers to Blender mesh data and modifies the appleseed MeshObject entity
+//      blender_object = the appleseed MeshObject created earlier in the export process
+//      vertices_length = the number of vertices in the mesh
+//      vertex_pointer = a string pointer to the first element of the vertex array
+//      Same template applies to mesh faces
+
+void convert_bl_mesh(
+    MeshObject*                 blender_mesh,
+    const size_t                vertices_length,
+    const size_t                vertex_pointer,
```

Comment:
The correct integer type to store pointer values is `uintptr_t`: https://stackoverflow.com/questions/1845482/what-is-uintptr-t-data-type